src/main/java/net/sf/jabref/gui/groups/TransferableEntrySelection.java

```diff
@@ -39,7 +39,8 @@
     public TransferableEntrySelection(List<BibEntry> list) {
         this.selectedEntries = list;
         selectedEntriesCiteKeys = String.join(",",
-                this.selectedEntries.stream().map(BibEntry::getCiteKey).collect(Collectors.toList()));
+                this.selectedEntries.stream().map(BibEntry::getCiteKeyOptional).filter(key -> key.isPresent())
```

Comment:
I agree my code looks weird, but I cannot figure it out. orElse won't work since we are operating on a stream (which happens to be a stream of optional). Based on the example here one can write it slightly more compact though. http://stackoverflow.com/questions/24254209/using-optional-with-streams-in-java